### DISCUSSION TITLE
Best place for global variables required by plugins

### DISCUSSION THREAD

**Discussion by kcaashish** [LazyVim/LazyVim#382] [Q&A] [answered]
I am using `fatih/vim-go` with `gopls` for definitions. So, as per the suggestions [here](https://github.com/golang/tools/blob/master/gopls/doc/vim.md#vim-go), I have added these lines into `config/options.lua`:

```lua
vim.g.go_def_mode = "gopls"
vim.g.go_info_mode = "gopls"
```

and have my plugin setup in `plugins/go.lua` as:

```lua
return {
  -- add the go plugin
  {
    "fatih/vim-go",
  },
}
```

It does the job, but I wanted to know if this is the way to do it. Is there a way I can define them inside go.lua?

**Comment by folke** [comment]
You can also add them insiede a `init()` function for the `vim-go` plugin

**Comment by loiccoyle** [comment, marked as answer]
You can put them in the `init` of the `vim-go` spec.
Something like:
```
return {
    "fatih/vim-go", 
    init=function()
        vim.g.go_def_mode = "gopls"
        vim.g.go_info_mode = "gopls"
    end
}
```

**Comment by kcaashish** [comment]
Thank you @folke @loiccoyle!

**Reply by loiccoyle** [reply]
Oh I didn't see folke got here first! You're welcome!
